mne/epochs.py

```diff
@@ -2492,8 +2493,158 @@ def _ensure_list(x):
     return metadata, events, event_id
 
 
+class AnnotationsMixin():
+    """Mixin class for Annotations in Epochs."""
+
+    @property
+    def annotations(self):  # noqa: D102
+        return self._annotations
+
+    @verbose
+    def _set_annotations(self, annotations, on_missing='raise'):
+        """Setter for Epoch annotations from Raw.
+
+        This private function simply copies over Raw annotations, and
+        does not handle offsetting the times based on first_samp
+        or measurement dates, since that is expected to occur in
+        Raw.set_annotations().
+
+        Parameters
+        ----------
+        annotations : instance of mne.Annotations | None
+            Annotations to set.
+        %(on_missing_ch_names)s
+
+        Returns
+        -------
+        self : instance of Raw
+            The raw object with annotations.
+        """
+        if annotations is None:
+            self._annotations = None
+        else:
+            _validate_type(annotations, Annotations, 'annotations')
```

Comment:
This will give a misleading message saying it has to be Annotations when it can also be None. Better to move it above the conditional with `(Annotations, None)` as the types

Reply:
This is the way it was done in Raw too. Should I change it there as well?